include/zephyr/bluetooth/addr.h

```diff
@@ -213,7 +214,7 @@ static inline bool bt_addr_le_is_identity(const bt_addr_le_t *addr)
  */
 static inline int bt_addr_to_str(const bt_addr_t *addr, char *str, size_t len)
 {
-	return snprintk(str, len, "%02X:%02X:%02X:%02X:%02X:%02X",
+	return snprintf(str, len, "%02X:%02X:%02X:%02X:%02X:%02X",
```

Comment:
You mention the reason for changing this from `snprintk` to `snprintf` was due to linker issues. Can you elaborate?

Reply:
we didn't want to pull in `log_minimal.c`, the original test had a copy-pasted `z_log_minimal_printk()` to avoid this.
We also had issues with `snprintk` pulling in additional depencies, like `cbvprintf`.
So we just thought it'd be okay to change it to use the libc rather than the kernel facilities. 

Do you know if there are any situations where one should prefer using `snprintk` over `snprintf`? we didn't think of any but we don't know the kernel that much, if you have any we can find an alternative to this change.

Reply:
I figured that in Zephyr context we would always use the `k` functions, to avoid any issues when changing stdlibs. 

If there are issues with `snprintk`, shouldn't they be solved, instead of just replacing them?

Reply:
One of the reasons for preferring snprintk in the past is that it has had a substantially smaller stack footprint.

Reply:
I also wanted to add that we should avoid linking two different formatters into the final image, if possible. I.e. either we should be using the printk formatter all across the board, or then use libc everywhere.

Reply:
> One of the reasons for preferring snprintk in the past is that it has had a substantially smaller stack footprint.

I think this is not true anymore, since the stack size didn't increase that much (80 bytes) when we switched to libc.

Reply:
I think that fixing the `snprintk` thing would be out of the scope of this PR. I reverted the change.